tests/components/withings/test_init.py

```diff
@@ -252,3 +177,28 @@ async def test_set_config_unique_id(
 
         await hass.config_entries.async_setup(config_entry.entry_id)
         assert config_entry.unique_id == "my_user_id"
+
+
+async def test_set_convert_unique_id_to_string(
+    hass: HomeAssistant, component_factory: ComponentFactory
+) -> None:
+    """Test upgrading configs to use a unique id."""
+    person0 = new_profile_config("person0", 0)
+
+    await component_factory.configure_component(profile_configs=(person0,))
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, data={"token": {"userid": 1234}, "profile": person0.profile},
+    )
+
+    with patch("homeassistant.components.withings.async_get_data_manager") as mock:
```

Comment:
Try to patch the library instead so we don't need to mock integration code.

Reply:
I'm not sure what you're asking me to patch. Please elaborate.

Reply:
The goal is to be able to set up the integration via the core without mocking any internals of the integration, like the data manager or update coordinator. To avoid I/O in the test I'm assuming we need to patch the library to prevent I/O from happening.